File: common/src/main/java/cc/turtl/chiselmon/feature/pc/bookmark/BookmarkConstants.java
package cc.turtl.chiselmon.feature.pc.bookmark;

import static cc.turtl.chiselmon.util.MiscUtil.modResource;

import cc.turtl.chiselmon.util.format.ComponentUtils;
import net.minecraft.client.gui.components.Tooltip;
import net.minecraft.resources.ResourceLocation;

/**
 * Configuration constants for PC Bookmark Buttons.
 */
public class BookmarkConstants {

    // Bookmark button
    public static final ResourceLocation BOOKMARK_SPRITE = modResource("textures/gui/pc/pc_button_bookmark.png");
    public static final int BOOKMARK_TEXTURE_WIDTH = 15;
    public static final int BOOKMARK_TEXTURE_HEIGHT = 30;
    public static final Tooltip BOOKMARK_ACTIVE_TOOLTIP =
            Tooltip.create(ComponentUtils.modTranslatable("pc.bookmark_button.tooltip.remove"));
    public static final Tooltip BOOKMARK_INACTIVE_TOOLTIP =
            Tooltip.create(ComponentUtils.modTranslatable("pc.bookmark_button.tooltip.add"));

    // Home button
    public static final ResourceLocation HOME_SPRITE = modResource("textures/gui/pc/pc_button_home.png");
    public static final int HOME_TEXTURE_WIDTH = 15;
    public static final int HOME_TEXTURE_HEIGHT = 30;
    public static final Tooltip HOME_TOOLTIP =
            Tooltip.create(ComponentUtils.modTranslatable("pc.home_button.tooltip"));

    // Tab button
    public static final ResourceLocation TAB_SPRITE = modResource("textures/gui/pc/pc_button_tab.png");
    public static final int TAB_TEXTURE_WIDTH = 35;
    public static final int TAB_TEXTURE_HEIGHT = 20;
    public static final int TAB_TEXT_MARGIN = 5;

    private BookmarkConstants() {
    }
}

File: common/src/main/kotlin/cc/turtl/chiselmon/client/feature/pc/bookmark/BookmarkConstants.kt
package cc.turtl.chiselmon.client.feature.pc.bookmark

import cc.turtl.chiselmon.core.util.modResource
import net.minecraft.client.gui.components.Tooltip
import net.minecraft.network.chat.Component

internal object BookmarkConstants {
    // Layout
    const val BOOKMARK_BUTTON_OFFSET_X = 239
    const val BOOKMARK_BUTTON_OFFSET_Y = 12
    const val HOME_BUTTON_OFFSET_X = 90
    const val HOME_BUTTON_OFFSET_Y = 12
    const val TAB_START_OFFSET_X = 80
    const val TAB_START_OFFSET_Y = -5
    const val TAB_HORIZONTAL_SPACING = 2

    // Bookmark button
    val BOOKMARK_SPRITE = modResource("textures/gui/pc/pc_button_bookmark.png")
    const val BOOKMARK_TEXTURE_WIDTH = 15
    const val BOOKMARK_TEXTURE_HEIGHT = 30
    val BOOKMARK_ACTIVE_TOOLTIP: Tooltip =
        Tooltip.create(Component.translatable("chiselmon.pc.bookmark_button.tooltip.remove"))
    val BOOKMARK_INACTIVE_TOOLTIP: Tooltip =
        Tooltip.create(Component.translatable("chiselmon.pc.bookmark_button.tooltip.add"))

    // Home button
    val HOME_SPRITE = modResource("textures/gui/pc/pc_button_home.png")
    const val HOME_TEXTURE_WIDTH = 15
    const val HOME_TEXTURE_HEIGHT = 30
    val HOME_TOOLTIP: Tooltip = Tooltip.create(Component.translatable("chiselmon.pc.home_button.tooltip"))

    // Tab button
    val TAB_SPRITE = modResource("textures/gui/pc/pc_button_tab.png")
    const val TAB_TEXTURE_WIDTH = 35
    const val TAB_TEXTURE_HEIGHT = 20
    const val TAB_TEXT_MARGIN = 5
}